pkg/gitops/profile.go

```diff
@@ -0,0 +1,196 @@
+package gitops
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"sigs.k8s.io/kustomize/pkg/gvk"
+	"sigs.k8s.io/kustomize/pkg/patch"
+	"sigs.k8s.io/kustomize/pkg/types"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	overlaysDir = "overlays/cluster-components"
+	baseDir     = "base"
+
+	defaultProfileGitHost = "github.com"
+	defaultProfileGitPath = "weaveworks/eks-gitops-example.git"
+)
+
+// GitOptions holds options for cloning a git repository
+type GitOptions struct {
+	URL    string
+	Branch string
+}
+
+// CloneFunc is a function that clones a Git repository
+type CloneFunc func(ctx context.Context, path string, options *git.CloneOptions) error
+
+// Profile represents a GitOps profile
+type Profile struct {
+	ClusterName string
+	Path        string
+	CloneFunc   CloneFunc
+	Fs          afero.Fs
+	IO          afero.Afero
+}
+
+type jsonPatch struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+// ManifestFile represents a manifest file with data
+type ManifestFile struct {
+	Name string
+	Data []byte
+}
+
+// Generate clones the specified Git repo in a base directory and generates overlays if the Git repo
+// points to a profile repo
+func (p *Profile) Generate(ctx context.Context, o GitOptions) error {
```

Comment:
Very nice separating this into a different package without dependencies to cobraCmd things